### PR TITLE
Kdr showdown

### PR DESCRIPTION
**Pull request by kdreese**
Added functionality for showdown game mode
 - Dropdown selector in lobby changes game mode
 - Players start with only one arrow
 - Game mode ends when only one team remains

TODO:
 - Remove friendly fire
 - Extend showdown to multiple rounds

### CODE REVIEW

File: src/autoload/multiplayer.gd [hunk 1]
```diff
@@ -65,7 +71,8 @@ class PlayerInfo:
 ## Game information, shared between players.
 class GameInfo:
 	var server_name: String = ""
-	var mode: GameMode = GameMode.FFA
+	var gamemode: GameMode = GameMode.SHOWDOWN
```
Comment (thearst3rd) [inline src/autoload/multiplayer.gd]: Nitpick, maybe `game_mode`?

File: src/autoload/multiplayer.gd [hunk 2]
```diff
@@ -65,7 +71,8 @@ class PlayerInfo:
...
+	var mode: TeamMode = TeamMode.FFA
```
Comment (thearst3rd) [inline src/autoload/multiplayer.gd]: This should be `team_mode` as to not get it confused with the game mode

File: src/states/game.gd [hunk 3]
```diff
@@ -219,6 +219,16 @@ func move_to_spawn_point(transform: Transform3D) -> void:
...
 
 
+# Get all targets not about to be deleted
+func get_targets() -> Array:
+	var targets := []
+	var all_targets := get_tree().get_nodes_in_group("Targets")
+	for target in all_targets:
+		if not target.is_queued_for_deletion():
+			targets.append(target)
+	return targets
```
Comment (thearst3rd) [inline src/states/game.gd]: Not at all required but we could be cool asf here
```suggestion
	return get_tree().get_nodes_in_group("Targets").filter(func(x): return not x.is_queued_for_deletion())
```
😩

File: src/states/game.gd [hunk 4]
```diff
@@ -73,6 +74,12 @@ func _ready() -> void:
 		Multiplayer.all_players_ready.connect(on_all_players_ready)
 	Multiplayer.rpc_id(1, "player_is_ready")
 
+	for player in Multiplayer.get_players():
+		if team_roster.keys().has(player.team_id):
```
Comment (admoore0) [inline src/states/game.gd]: `player.team_id in team_roster` is a bit cleaner IMO

File: src/states/game.gd [hunk 5]
```diff
@@ -244,24 +254,22 @@ func everyone_gets_an_arrow(id: String, power: float) -> void:
 		return
 	var player := $Players.get_node(id)
 	if player.state == Player.PlayerState.NORMAL and player.num_arrows > 0: # if player meets the requirements to be able to shoot
-		spawn_arrow.rpc(id, power)
-		player.num_arrows -= 1
-		rpc_id(int(id), "update_quiver_amt", player.num_arrows)
+		rpc("spawn_arrow", id, power)
```
Comment (admoore0) [inline src/states/game.gd]: `spawn_arrow.rpc(...)`

File: src/states/game.gd [hunk 6]
```diff
@@ -198,12 +192,18 @@ func assign_spawn_point(player_id: int) -> void:
 		spawn_point = available_spawn_points.pick_random() as SpawnPoint
 	spawn_point.assigned_player_id = player_id
 	rpc_id(player_id, "move_to_spawn_point", spawn_point.transform)
+	rpc("respawn_player", player_id)
```
Comment (thearst3rd) [inline src/states/game.gd]:
```suggestion
	respawn_player.rpc(player_id)
```

File: src/states/game.gd (hunk 5 above)
Comment (thearst3rd) [inline src/states/game.gd]: burh change dis back
```suggestion
		spawn_player.rpc(id, power)
```

File: src/states/showdown_gamemode.gd [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+	var targets := get_targets()
+	for target in targets:
+		target.queue_free()
+
+
+func _physics_process(delta: float) -> void:
+	super._physics_process(delta)
+	if game_state == GameState.ENDED:
+		match_timer.text = "Time's up!"
+		return
```
Comment (admoore0) [inline src/states/showdown_gamemode.gd]: Delete or change text

**Review by admoore0 — CHANGES_REQUESTED** [review]

**Review by thearst3rd — CHANGES_REQUESTED** [review]
pit nicks

File: src/states/menus/lobby.gd [hunk 8]
```diff
@@ -125,11 +125,11 @@ func start_game() -> void:
 # :param new_mode_id: The ID of the selected mode.
 func on_mode_select(new_mode_id: Multiplayer.TeamMode) -> void:
```
Comment (admoore0) [inline src/states/menus/lobby.gd]: on_team_mode_select?

**Review by admoore0 — APPROVED** [review]

File: src/states/game.gd [hunk 9]
```diff
@@ -71,11 +71,11 @@ func _ready() -> void:
 	Multiplayer.server_disconnected.connect(server_disconnected)
 
 	if is_multiplayer_authority():
-		Multiplayer.all_players_ready.connect(on_all_players_ready)
+		Multiplayer.all_players_loaded.connect(on_all_players_ready)
```
Comment (thearst3rd) [inline src/states/game.gd:74]: fixing andrews heck up 😈😈😈😈😈😈

**Review by thearst3rd — APPROVED** [review]

**Pull request merged**